anet-tools/tests/integration/test_insert_update_nested_entity.py

```diff
@@ -0,0 +1,234 @@
+        self.update_rules["tables"].append(
+            {"name": "locations", "columns": ["name"]})
+        base_methods.is_entity_update(position, self.update_rules)
+        base_methods.is_entity_update(person, self.update_rules)
+        base_methods.is_entity_update(location, self.update_rules)
+
+        position.insert_update_nested_entity(self.update_rules, self.utc_now)
+
+        pp_q = self.session.query(self.PeoplePositions).filter(
+            self.PeoplePositions.personUuid == person.uuid, self.PeoplePositions.positionUuid == position.uuid).all()
+
+        result = len(pp_q)
+        expected = 1
+        self.assertEqual(result, expected)
+
+        result = pp_q[0].position.currentPersonUuid
+        expected = person.uuid
+        self.assertEqual(result, expected)
+
+        result = pp_q[0].position.locationUuid
+        expected = location.uuid
+        self.assertEqual(result, expected)
+
+    def test_insert_update_nested_entity_people_update_pos_update_case2(self):
+        print("Tests: Update Position (has no former person association) Update Person (has former position association)")
+
+        position = self.Position(
+            name="EF 1.1 Old Inactive Advisor", type=0, status=0)
+        person = self.Person(name="HENDERSON, Henry", role=1)
+        location = self.Location(name="Wishingwells Park", status=1)
+
+        position.person = person
+        position.location = location
+
+        self.update_rules["tables"].append(
+            {"name": "positions", "columns": ["name"]})
+        self.update_rules["tables"].append(
+            {"name": "people", "columns": ["name"]})
+        self.update_rules["tables"].append(
+            {"name": "locations", "columns": ["name"]})
+        base_methods.is_entity_update(position, self.update_rules)
+        base_methods.is_entity_update(person, self.update_rules)
+        base_methods.is_entity_update(location, self.update_rules)
+
+        formerPosUuid = self.session.query(self.Position).filter(
+            self.Position.currentPersonUuid == person.uuid).all()[0].uuid
+
+        position.insert_update_nested_entity(self.update_rules, self.utc_now)
+
+        pp_q = self.session.query(self.PeoplePositions).filter(
+            self.PeoplePositions.personUuid == person.uuid, self.PeoplePositions.positionUuid == position.uuid).all()
+
+        result = len(pp_q)
+        expected = 1
+        self.assertEqual(result, expected)
+
+        result = pp_q[0].position.currentPersonUuid
+        expected = person.uuid
+        self.assertEqual(result, expected)
+
+        result = pp_q[0].position.locationUuid
+        expected = location.uuid
+        self.assertEqual(result, expected)
+
+        pp_q = self.session.query(self.PeoplePositions).filter(self.PeoplePositions.personUuid == None,
+                                                               self.PeoplePositions.positionUuid == formerPosUuid, self.PeoplePositions.endedAt == None).all()
```

Comment:
Instead of these long lines, you can separate the method chains like this:
```
        pp_q = self.session.query(self.PeoplePositions) \
                    .filter(self.PeoplePositions.personUuid == None,
                            self.PeoplePositions.positionUuid == formerPosUuid,
                            self.PeoplePositions.endedAt == None) \ 
                    .all()
```